custom_components/nhc2/entities/generic_hvac_climate.py

```diff
@@ -147,13 +176,31 @@ async def async_set_temperature(self, **kwargs):
 
     async def async_set_hvac_mode(self, hvac_mode: str):
         if hvac_mode == HVACMode.OFF:
-            self._device.set_status(self._gateway, PROPERTY_STATUS_VALUE_OFF)
+            self._device.set_status(self._gateway, False)
+
+        # switch on the device if it is off
+        if hvac_mode != HVACMode.OFF and not self._device.is_status_on:
+            self._device.set_status(self._gateway, True)
+
         if hvac_mode == HVACMode.HEAT:
             self._device.set_operation_mode(self._gateway, PROPERTY_OPERATION_MODE_VALUE_HEAT)
         if hvac_mode == HVACMode.COOL:
             self._device.set_operation_mode(self._gateway, PROPERTY_OPERATION_MODE_VALUE_COOL)
         if hvac_mode == HVACMode.AUTO or hvac_mode == HVACMode.HEAT_COOL:
             self._device.set_operation_mode(self._gateway, PROPERTY_OPERATION_MODE_VALUE_AUTO)
+        if hvac_mode == HVACMode.FAN_ONLY:
+            self._device.set_operation_mode(self._gateway, PROPERTY_OPERATION_MODE_VALUE_FAN)
+        if hvac_mode == HVACMode.DRY:
+            self._device.set_operation_mode(self._gateway, PROPERTY_OPERATION_MODE_VALUE_DRY)
+
+        self.on_change()
+
+    async def async_turn_on(self) -> None:
```

Comment:
I don't think these methods are supported by Home Assistant. As I can't find them in the documentation for the Climate Entity: https://developers.home-assistant.io/docs/core/entity/climate/#methods